src/Microsoft.ML.Core/Environment/ConsoleEnvironment.cs

```diff
@@ -15,7 +13,12 @@ namespace Microsoft.ML.Runtime.Data
 {
     using Stopwatch = System.Diagnostics.Stopwatch;
 
-    public sealed class ConsoleEnvironment : HostEnvironmentBase<ConsoleEnvironment>
+    /// <summary>
+    /// The console environment. As its name suggests, should be limited to those applications that deliberately want
+    /// console functionality.
+    /// </summary>
+    [BestFriend]
+    internal sealed class ConsoleEnvironment : HostEnvironmentBase<ConsoleEnvironment>
```

Comment:
I believe we can close #1284 with this change.

Reply:
OK. Added to the description.

---
In reply to: [234249735](https://github.com/dotnet/machinelearning/pull/1626#discussion_r234249735) [](ancestors = 234249735)